internal/controller/ols_app_server_assets.go

```diff
@@ -50,10 +50,30 @@ func (r *OLSConfigReconciler) generateOLSConfigMap(cr *olsv1alpha1.OLSConfig) (*
 		}
 		providerConfigs = append(providerConfigs, providerConfig)
 	}
-
-	conversationCache := ConversationCacheConfig{
-		Type:   string(cr.Spec.OLSConfig.ConversationCache.Type),
-		Memory: MemoryCacheConfig{MaxEntries: cr.Spec.OLSConfig.ConversationCache.Memory.MaxEntries},
+	var conversationCache ConversationCacheConfig
+	if cr.Spec.OLSConfig.ConversationCache.Memory != (olsv1alpha1.MemorySpec{}) {
+		conversationCache = ConversationCacheConfig{
+			Type:   string(cr.Spec.OLSConfig.ConversationCache.Type),
+			Memory: MemoryCacheConfig{MaxEntries: cr.Spec.OLSConfig.ConversationCache.Memory.MaxEntries},
+		}
+	} else {
+		var redisCredentialsConfig RedisCredentialsConfig
+		if cr.Spec.OLSConfig.ConversationCache.Redis != (olsv1alpha1.RedisSpec{}) && cr.Spec.OLSConfig.ConversationCache.Redis.Credentials != (olsv1alpha1.RedisCredentialsSpec{}) {
+			redisCredentialsConfig = RedisCredentialsConfig{
+				UsernamePath: cr.Spec.OLSConfig.ConversationCache.Redis.Credentials.UsernamePath,
+				PasswordPath: cr.Spec.OLSConfig.ConversationCache.Redis.Credentials.PasswordPath,
+			}
+		}
+		conversationCache = ConversationCacheConfig{
```

Comment:
@vishnuchalla If we dont have anything defined we can hit the below lines of code , to get ` cr.Spec.OLSConfig.ConversationCache.Redis` ? Please check.

Reply:
Modified this logic and verified the working of all those cases.

Reply:
@vishnuchalla Looks good . Could you please add the tests for this PR . I don't think we have a separate story for tests.

Reply:
@xrajesh IMHO, I would want to take unit tests in a separate story: https://issues.redhat.com/browse/OLS-333, to make the review process smoother.

Reply:
Thanks - I am okay with separate story- we we need to get that as priority. I will approve when the redis image is ready.

Reply:
Acknowledged. And the image is available for use. Verified it by testing in local.

Reply:
@xrajesh just a gentle reminder on this.